salt/proxy/junos.py

```diff
@@ -101,6 +110,23 @@ def init(opts):
     if 'username' in opts['proxy'].keys():
         opts['proxy']['user'] = opts['proxy'].pop('username')
     proxy_keys = opts['proxy'].keys()
+
+    # If encoded_passwd is found, prefer it over passwd or password
+    if 'encoded_password' in opts['proxy'].keys():
+        if HAS_JUNOSSECURE:
+            try:
+                decoded_password = junos_decode(opts['proxy'].pop('encoded_password'))
+                opts['proxy'].pop('passwd', None)
+                opts['proxy'].pop('password', None)
+                opts['proxy']['passwd'] = decoded_password
+            except EncodeDecodeError:
+                log.error('Unable to decode encoded_password, proceeding with passwd or password')
+    # Proceeding with plain text password options as junossecure package not found
+        else:
+            opts['proxy'].pop('encoded_password')
+            log.error('encoded_password option provided, but could not find junossecure option to'
+                      'decode. Proceeding with passwd or password options if provided.')
```

Comment:
Need a space before decode, otherwise message will to displayed as '... option todecode ..' Minor one please fix it.

Reply:
Done